deployment/terraform/modules/s3/outputs.tf

```diff
@@ -1,3 +1,7 @@
 output "upload_bucket_name" {
   value = "${aws_s3_bucket.uploaded_files.id}"
 }
+
+output "bucket_name_base" {
```

Comment:
`upload_bucket_name` is derived from `bucket_name_base`, so having both is redundant

Reply:
There's nothing unusual about deriving a value and storing it under a new name. I'm not sure what your concern here is.

in s3/main.tf:
```
bucket = "${var.bucket_name_base}-upload"
```

s3/outputs.tf:
```
output "upload_bucket_name" {
  value = "${aws_s3_bucket.uploaded_files.id}"
}
```

cognito/main.tf:
```
      "Resource": "arn:aws:s3:::${var.upload_bucket_name}/$${cognito-identity.amazonaws.com:sub}/*"
```

Remembering that  the way we form the name is to append "-upload", and doing that in two widely separated places does not feel like a better alternative. Also: If we stop creating that bucket for some reason, the downstream references will also break when terraform tries to construct the plan, rather than just relying on naming conventions being consistent.

Reply:
`S3_BUCKET_NAME_BASE` is set in `config.yaml` then it is used as the name of the Terraform workspace and finally as an input parameter to the S3 module (`bucket_name_base`). There is no need to output this value to include back in `config.yaml` since it already exists there.

Reply:
I think that when we looked at this we thought it was odd that we had to set `S3_BUCKET_NAME_BASE` in config.yaml and then be sure to provide the same value as the terraform workspace name. 

Wouldn't it make sense to set this value once upon workspace creation and have `S3_BUCKET_NAME_BASE` be updated automatically?

Reply:
There is no need to automate creation of `config.yaml` because it is going away after we switch the rest of the infrastructure from CloudFormation to Terraform.

Reply:
is this blocking?

Reply:
It's easier for me at least to think about this if information can flow in one direction, rather that starting in config.yaml, then terraform, then back to config.yaml.

I've run this a couple times, and I feel much more confident in procedures which minimize manual intervention. If we can throw out the ERB in a week or two, great.

The goal we're aiming for is to have it start (and end) with terraform, so this feels like the right direction.

If this is a blocking issue for you, please tell precisely what you want reverted so we can just do it.

Reply:
OK. I've created PR #2683 to show the proposed changes.